Document expected delay on CNAME change and need for CAA

nijel opened the pull request:
The SSL certificate issuing takes about one hour in this case according to comment in #2652.

davidfischer commented:
This is only the case for users who had an existing custom domain setup and then corrected their setup to comply with the [custom domains docs](https://docs.readthedocs.io/en/latest/alternate_domains.html#cname-support). This doesn't apply to new setups. Perhaps there's a way to make that clear.

nijel commented:
Also one hour apparently is not enough. It took more than 12 hours with docs.weblate.org, I've switched docs.phpmyadmin.net about two hours ago and still not working...

davidfischer commented:
> Also one hour apparently is not enough. It took more than 12 hours with docs.weblate.org

Did you have a CAA record on that domain? It may have been due to that.

> docs.phpmyadmin.net

I don't even see that domain in the list of certificates that are pending. I see rtd-backend.phpmyadmin.net, but not docs.phpmyadmin.net. It is possible since we switched over a few weeks ago that the request to issue the certificate timed out. If you save the domain record in the Read the Docs UI (`https://readthedocs.org/dashboard/<slug>/domains/`), it should trigger adding the certificate again.

The real answer here is I need to get this status raised up to users in the UI.

nijel commented:
It was already there, but I've removed and added it back now. And it now works.

8. srpna 2018 17:52:23 SELČ, David Fischer <notifications@github.com> napsal:
>> Also one hour apparently is not enough. It took more than 12 hours
>with docs.weblate.org
>
>Did you have a CAA record on that domain? It may have been due to that.
>
>> docs.phpmyadmin.net
>
>I don't even see that domain in the list of certificates that are
>pending. I see rtd-backend.phpmyadmin.net, but not docs.phpmyadmin.net.
>It is possible since we switched over a few weeks ago that the request
>to issue the certificate timed out. If you save the domain record in
>the Read the Docs UI
>(`https://readthedocs.org/dashboard/<slug>/domains/`), it should
>trigger adding the certificate again.
>
>The real answer here is I need to get this status raised up to users in
>the UI.

nijel commented:
And yes I did have CAA on both domains, but it was properly adjusted so it should not cause additional issues.

8. srpna 2018 17:52:23 SELČ, David Fischer <notifications@github.com> napsal:
>> Also one hour apparently is not enough. It took more than 12 hours
>with docs.weblate.org
>
>Did you have a CAA record on that domain? It may have been due to that.
>
>> docs.phpmyadmin.net
>
>I don't even see that domain in the list of certificates that are
>pending. I see rtd-backend.phpmyadmin.net, but not docs.phpmyadmin.net.
>It is possible since we switched over a few weeks ago that the request
>to issue the certificate timed out. If you save the domain record in
>the Read the Docs UI
>(`https://readthedocs.org/dashboard/<slug>/domains/`), it should
>trigger adding the certificate again.
>
>The real answer here is I need to get this status raised up to users in
>the UI.

davidfischer commented on docs/alternate_domains.rst:
```diff
@@ -52,6 +52,8 @@ This service is generously provided by Cloudflare.
     they do not yet allow us to acquire SSL certificates for those domains.
     Simply point the CNAME to ``readthedocs.io``.
+    
+    If you change the CNAME, the SSL certificate issuance will take about one hour.
```
nitpick: I prefer "can take" rather than "will take" especially since for a new CNAME (as opposed to a changed one) this should be almost immediate.

nijel commented:
I've added CAA docs and adjusted wording based on your suggestion.

davidfischer commented:
Thanks. Looks great!
